torch/_meta_registrations.py

```diff
@@ -2426,8 +2426,14 @@ def meta_sort(self, stable=None, dim=-1, descending=False, values=None, indices=
     if values is not None and indices is not None:
         assert isinstance(values, TensorLike)
         assert isinstance(indices, TensorLike)
-        values = _maybe_resize_out(values, v.shape)
-        indices = _maybe_resize_out(indices, i.shape)
+        # Makes sure values and indices have the same strides. For cases where
+        # these have different shapes, like (5, 10, 5) and (0) in msort.
+        out_shape = v.shape
+        out_stride = v.stride()
+        values = _maybe_resize_out(values, out_shape)
+        indices = _maybe_resize_out(indices, out_shape)
+        values.as_strided_(out_shape, out_stride)
+        indices.as_strided_(out_shape, out_stride)
```

Comment:
Heads up: pushed a fix for msort, due to:

```cpp
Tensor& msort_out(const Tensor& self, Tensor& values) {
  Tensor indices = at::empty({0}, self.options().dtype(kLong));
  at::sort_out(values, indices, self, 0, false);
  return values;
}
```

Reply:
This logic is not correct. The logic in core is given by
```python
  // See issue: https://github.com/pytorch/pytorch/issues/65863
  // Strides should be dense, so as not to allocate too much memory.
  // We either use 'self' strides, or infer dense strides from them.
  std::vector<int64_t> strides = (self.is_non_overlapping_and_dense())
      ? self.strides().vec()
      : at::infer_dense_strides(self.sizes(), self.strides());

  set_output_raw_strided(0, self.sizes(), strides, self.options(), {});
  set_output_raw_strided(1, self.sizes(), strides, self.options().dtype(kLong), {});
```